src/mlpack/methods/linear_svm/linear_svm.hpp

```diff
@@ -264,6 +270,26 @@ class LinearSVM
   bool fitIntercept;
 };
 
+/**
+ * Callback test function, based on the EndOptimization callback function.
+ */
+class CallbackTestFunction
+{
+ public:
+    CallbackTestFunction() : calledEndOptimization(false) {}
+
+    template<typename OptimizerType, typename FunctionType, typename MatType>
+    void EndOptimization(OptimizerType& /* optimizer */,
+                       FunctionType& /* function */,
+                       MatType& /* coordinates */)
+    {
+        calledEndOptimization = true;
+    }
+
+    //! Track to check if callback is executed.
+    bool calledEndOptimization;
+};
```

Comment:
Can we put this in the test code directory?

Reply:
@rcurtin Should I create a new file as "linear_svm_test_tools.hpp" and place the           ```class CallbackTestFunction``` under it ?

Reply:
I think it would be fine to just put the class definition in `linear_svm_test.cpp`.

Reply:
Test comment for mlpack-bot...

Reply:
Test comment for mlpack-bot...

Reply:
Test comment for mlpack-bot...

Reply:
Test comment for mlpack-bot...